blocks/form/components/repeat/repeat.js

```diff
@@ -88,42 +107,262 @@ function getInstances(el) {
+              // eslint-disable-next-line max-len
+              // Reason for requestAnimationFrame: Model changes fire immediately but
+              // DOM updates are async.
+              // We need to wait for the browser's next paint cycle
+              // to ensure the new/removed fieldsets
+              // are in the DOM before adding/updating buttons.
+              requestAnimationFrame(() => {
+                // eslint-disable-next-line no-use-before-define
+                addRemoveButtons(wrapper, form, repeatStrategies.af);
+                updateRepeatState(wrapper);
+              });
+            }
+          });
+        }, 'change');
+      });
+    },
+  },
+
+  doc: {
+    /**
+     * Adds a new instance using direct DOM manipulation (document-based forms).
+     * @param {HTMLElement} wrapper - The repeat wrapper element
+     * @param {HTMLElement} form - The form element
+     */
+    addInstance: (wrapper, form) => {
+      const fieldset = wrapper['#repeat-template'];
+      const childCount = wrapper.children.length - 1;
+      const newFieldset = fieldset.cloneNode(true);
+      newFieldset.setAttribute('data-index', childCount);
+      update(newFieldset, childCount, wrapper['#repeat-template-label']);
+      updateSelectFieldNames(newFieldset, childCount);
+
+      const actions = wrapper.querySelector('.repeat-actions');
+      actions.insertAdjacentElement('beforebegin', newFieldset);
+
+      // Add remove button to the new instance
+      // eslint-disable-next-line no-use-before-define
+      insertRemoveButton(newFieldset, wrapper, repeatStrategies.doc);
+
+      // Add remove buttons to all existing instances that don't have them
+      // (this handles the case where we started with min instances and no buttons)
+      // eslint-disable-next-line no-use-before-define
+      addRemoveButtons(wrapper, form, repeatStrategies.doc);
+
+      updateRepeatState(wrapper);
+
+      const event = new CustomEvent('item:add', {
+        detail: { item: { name: newFieldset.name, id: newFieldset.id } },
+        bubbles: false,
+      });
+      form.dispatchEvent(event);
+    },
+
+    /**
+     * Removes an instance using direct DOM manipulation (document-based forms).
+     * @param {HTMLElement} fieldset - The fieldset to remove
+     * @param {HTMLElement} wrapper - The repeat wrapper element
+     * @param {HTMLElement} form - The form element
+     */
+    removeInstance: (fieldset, wrapper) => {
+      fieldset.remove();
+      wrapper.querySelectorAll('[data-repeatable="true"]').forEach((el, index) => {
+        update(el, index, wrapper['#repeat-template-label']);
+      });
+      wrapper.querySelectorAll('[data-repeatable="true"]').forEach((el, index) => {
+        updateSelectFieldNames(el, index);
+      });
+      updateRepeatState(wrapper);
+    },
+
+    /**
+     * No setup needed for document-based forms.
+     */
+    setup: () => {
+      // No setup required for doc-based forms
+    },
```

Comment:
You can remove this function and add a check if the function exists before calling